pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -118,36 +121,45 @@ func listWatcher(client *dynamic.Client, resource schema.GroupVersionResource) *
 }
 
 func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind schema.GroupVersionKind) (cache.Controller, error) {
-	// TODO: consider store in one storage.
-	glog.V(5).Infof("create storage for resource %s", resource)
-	client, err := gb.metaOnlyClientPool.ClientForGroupVersionKind(kind)
-	if err != nil {
-		return nil, err
-	}
-	gb.registeredRateLimiterForControllers.registerIfNotPresent(resource.GroupVersion(), client, "garbage_collector_monitoring")
-	setObjectTypeMeta := func(obj interface{}) {
+	setObjectTypeMeta := func(obj interface{}, clone bool) (runtime.Object, error) {
 		runtimeObject, ok := obj.(runtime.Object)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("expected runtime.Object, got %#v", obj))
+			return nil, fmt.Errorf("expected runtime.Object, got %#v", obj)
+		}
+		if clone {
+			copy, err := api.Scheme.DeepCopy(runtimeObject)
```

Comment:
It's presumably going to do memory allocations and take some cpu time that it wouldn't otherwise be doing. I haven't measured it yet. I wanted to get this up for review first.

I just looked a bit more, and it only uses the object so it can get access to the `meta.Accessor` and `meta.TypeAccessor`. Maybe we could be more efficient and not use the full object?